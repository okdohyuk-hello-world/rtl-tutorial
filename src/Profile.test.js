import React from 'react';
import {render, screen} from "@testing-library/react";
import Profile from "./Profile";

describe('<Profile />', () => {
    it('matches snapshot', () => {
        const utils = render(<Profile username="okdohyuk" name="유도혁" />);
        expect(utils.container).toMatchSnapshot();
    });
    it('shows the props correctly', () => {
        render(<Profile username="okdohyuk" name="유도혁" />);
        screen.getByText('okdohyuk'); // okdohyuk 라는 텍스트를 가진 엘리먼트가 있는지 확인
        screen.getByText('(유도혁)'); // (유도혁) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
        screen.getByText(/유/); // 정규식 /유/ 을 통과하는 엘리먼트가 있는지 확인
    });
});
