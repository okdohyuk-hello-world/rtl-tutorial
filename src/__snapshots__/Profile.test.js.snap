// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      okdohyuk
    </b>
     
    <span>
      (
      유도혁
      )
    </span>
  </div>
</div>
`;
